src/sidebar/components/Annotation/AnnotationBody.tsx

```diff
@@ -71,6 +72,13 @@ function AnnotationBody({ annotation, settings }: AnnotationBodyProps) {
   const defaultAuthority = store.defaultAuthority();
   const draft = store.getDraft(annotation);
   const mentionsEnabled = store.isFeatureEnabled('at_mentions');
+  const mentionMode = useMemo(
+    (): MentionMode =>
+      isThirdPartyUser(annotation.user, defaultAuthority)
+        ? 'display-name'
+        : 'username',
+    [annotation.user, defaultAuthority],
+  );
```

Comment:
This logic is duplicated in `AnnotationEditor`. I'm already investigating in parallel what's the `Annotation` component tree, and how mention-related info is used in there. We may be able to reduce duplication and prop drilling if we use a mentions-scoped context.